drivers/can/can_stm32.h

```diff
@@ -56,15 +56,15 @@ enum can_filter_type {
 };
 
 struct can_stm32_data {
-	struct k_mutex tx_mutex;
-	struct k_mutex set_filter_mutex;
+	struct k_mutex inst_mutex;
```

Comment:
Looks like you have opted for a single mutex, maybe that makes sense for the mcp2515 driver too now with a growing data struct for the instances

Reply:
Use a global mutex for MCP2515 now. It saves memory and should not block too much.